gap/matrix/classical.gi

```diff
@@ -2257,15 +2257,8 @@ function( arg )
                    isSUContained := "unknown",
                    isSOContained := "unknown",
                   );
-  merkinfolevel := InfoLevel(InfoMethSel);
-  SetInfoLevel(InfoMethSel,0);
-  ret := CallMethods( ClassicalMethDb, nrrandels, recognise, grp );
-  SetInfoLevel(InfoMethSel,merkinfolevel);
-  # fail: bedeutet, dass entnervt aufgegeben wurde
-  # true: bedeutet, dass eine Methode "erfolgreich" war
-
+  CallMethods( ClassicalMethDb, nrrandels, recognise, grp );
```

Comment:
The commit "matrix: clean up RecogniseClassical" is mislabeled; it drops a change to the info level.  I wouldn't call this a "clean up" (for me, a "clean up" is a change that does not change observable behavior, just restructures the code).

As to the change itself: I am not yet sure whether it's a great idea or a horrible one... the current code certainly is a hack, though, because it (ab?)uses `CallMethods`; but all the "methods" that are installed there still return values like `fail` and `false` and `true` (and I am not even sure with the correct semantics!).

Reply:
It's also interesting to see that the return value of `CallMethods` is completely ignored.

Reply:
>  I wouldn't call this a "clean up" (for me, a "clean up" is a change that does not change observable behavior, just restructures the code).

Good to know. I also included changing the debugging output into "clean up".

> As to the change itself: I am not yet sure whether it's a great idea or a horrible one...

I'm a bit confused. I only changed the infolevels and got rid of the unused `ret` variable. Surely, only changing the infolevels can't be that bad. Or am I missing something?

I guess when I get to the classical groups I will try to reorganize the code and get rid of that `CallMethods` call.

Reply:
I mean, all we would possibly do is print _more_ information. Ofc I could add something like `Info(InfoMethsel, 2, "HACK: CallMethods with ClassicalMethDb");`

Reply:
Exactly, we'd be printing more information, potentially *lots* more information that one can not suppress anymore now, without also suppressing information about the "top level" method selection: because here a leaf recognition method uses `CallMethods` itself, which no other method does.

So I am concerned that some debugging might get considerably harder because the information one is interested in is hidden in a flood of info messages that weren't there before, and which I can't turn off without also turning off the messages I *am* interested in...

Reply:
My thinking was that since `RecogniseGeneric` calls itself recursively, it calls `CallMethods` on every node of the "recognition tree" anyways.
I guess it makes more sense for me to look at this again when I have some concrete examples or if I'm rewriting that code anyway. I'll drop the respective commit.

Reply:
I've moved the respective commit into PR #187